src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/StackSpiller.Temps.cs

```diff
@@ -219,4 +225,33 @@ internal void VerifyTemps()
             }
         }
     }
+
+    /// <summary>
+    /// An extension expression node used to obtain a reference to the
+    /// specified operand, much like `ref x` in C#. This node is used by
+    /// the stack spiller when spilling value type instances or arguments
+    /// passed to ref parameters.
+    /// </summary>
+    internal sealed class RefExpression : Expression
```

Comment:
Another approach, that might avoid creating a new node, could be adding a IsByRef bool flag to Assignment node.

That is - if assignment is ByRef, then an address of RHS is taken and assigned to the LHS that must be a byref local.
If assignment is regular, and LHS is a ref local - just do an indirect assignment. (not sure if the case will arise)

ByRef assignment can itself be used in Emit Expression and EmitAddrerss., if such need arises.
ByRef assignment is essentially a duplicated RHS address, so it can be used as both value and address. - I.E. ref assignment can itself ve a target of a byval assignment.

I think that could be simpler than introducing an extension node.